rest-api/dao/base_dao.py

```diff
@@ -157,25 +157,24 @@ def query(self, query_def):
     with self.session() as session:
       query, field_names = self._make_query(session, query_def)
       items = query.all()
-    if items:
-      if len(items) > query_def.max_results:
-        # Items, pagination token, and more are available
-        return Results(items[0:query_def.max_results],
-                       self._make_pagination_token(items[query_def.max_results - 1].asdict(),
-                                                   field_names),
-                       True)
-      else:
-        if query_def.always_return_token:
-          # Items and pagination token, but no more available
-          return Results(items,
-                         self._make_pagination_token(items[len(items) - 1].asdict(), field_names),
-                         False)
-        else:
-          # Items but no pagination token, and no more available
-          return Results(items, None, False)
-    else:
+      # We could use query.count() here but since we're executing and iterating the entire result
+      # set every time anyways it should be unnecessary: len(items) ought to be the same thing
+      total = len(items)
```

Comment:
That isn't correct. We limit results, based on a _count parameter or DEFAULT_MAX_RESULTS (100), here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/api/base_api.py#L116

That's the whole reason we have pagination tokens, too.

So you'll get at most pageSize + 1 results here. To get a total, you will have to run a separate count(*) query. In order to ensure this doesn't hurt performance in cases where we don't need the total, we should make this logic conditional upon the client asking for the total via some new query param (e.g. "_getTotal"=true)

If the client *does* ask for a total, we should always return it, even when there is no pagination token returned.